pkg/datapath/iptables/iptables.go

```diff
@@ -336,7 +336,6 @@ func egressProxyRule(cmd, l4Match, markMatch, mark, port, name string) []string
 	return []string{
 		"-t", "mangle",
 		cmd, ciliumPreMangleChain,
-		"-i", option.Config.EndpointInterfaceNamePrefix,
```

Comment:
Shouldn't we still keep the rule as is if not running in flannel mode?

Reply:
There's two broad approaches to this: Either parameterize everything so they run differently in flannel vs other modes, or try to run as close as you can to the same configuration in all modes. I opted for the latter, partly so that ideally we find out if there's a problem with this kind of configuration in any mode (rather than only in the more obscure flannel mode), and partly also because in other iptables rules we've been removing the device matches recently so it's more consistent with those rules.

There's still matches on the magic skb marks in these rules so I'm not too worried about them applying to traffic that we don't intend them to apply to.

Reply:
This was the only use for `EndpointInterfaceNamePrefix` so we can hide the option.